### PR TITLE
Expose TaskPathLocation as contextual locations on Tooling API events

### PR DESCRIPTION
**Pull request by donat**
Fixes https://github.com/gradle/gradle/issues/31553

When we introduced origin and contextual locations, we did not adjust how we expose that to the TAPI clients. Adding TaskPathLocation data to problem reports then went to the incorrect bucket (to the origin location bucket). 

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy [hunk 1]
```diff
@@ -24,7 +25,7 @@ import org.gradle.integtests.fixtures.GroovyBuildScriptLanguage
 
 import static org.gradle.api.problems.ReportingScript.getProblemReportingScript
 
-class ProblemsServiceIntegrationTest extends AbstractIntegrationSpec {
+class   ProblemsServiceIntegrationTest extends AbstractIntegrationSpec {
```
Comment (reinsch82) [inline platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy:28]: 💄 space

**Comment by donat** [comment]
@bot-gradle test this

File: platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy (hunk 1 above)
Comment (donat) [reply platforms/ide/problems-api/src/integTest/groovy/org/gradle/api/problems/ProblemsServiceIntegrationTest.groovy]: Fixed

**Comment by donat** [comment]
@bot-gradle test RFR

**Comment by donat** [comment]
@bot-gradle merge

**Pull request merged**